tests/components/template/test_cover.py

```diff
@@ -360,8 +484,8 @@ async def test_template_state_text_with_position(
     [
         (
             1,
-            "{{ states.cover.test_state.state }}",
-            "{{ states.cover.test_position.attributes.position }}",
+            "{{ states('cover.test_state') }}",
+            "{{ state_attr('cover.test_position', 'position') }}",
```

Comment:
Why is this changed?

Reply:
The templates purposely error because the state object doesn't exist.  I'll change it back

Reply:
Alright, after diving into this deeper.  The errors that are produced for non-trigger-based template entities occur in the caplog_setup_text, so they are missed when checking against `caplog.text`.  However with trigger based entities, the templates aren't resolved at startup/setup, they are resolved when the trigger occurs.  Which makes those errors appear in `caplog.txt`.  Unfortunately, I have no idea what error this is trying to capture because the original test only had "ERROR".  I'm thinking we should just remove the assert "ERROR" not in caplog.text and only verify the state is what we expect.

Reply:
The assert is checking that there is NOT an error.
it was added by @jbouwh in PR https://github.com/home-assistant/core/pull/91172, maybe he can enlighten us.